hera_cal/io.py

```
@@ -864,6 +864,147 @@ def iterate_over_times(self, Nints=1, times=None):
+        verbose: print some progress messages.
+
+    Returns:
+        rv: dictionary with keys 'info' (metadata), 'visdata' (dictionary of waterfalls
+            with i,j,pol keys), 'flags', and 'nsamples'. Will omit keys according to
+            read_data, read_flags, and read_nsamples.
+    '''
+    info = {}
+    times = []
+    bl2ind = {}
+    inds = {}
+    # Read file metadata to size up arrays and sort times
+    for filename in filenames:
+        if verbose:
+            print(f'Reading header of {filename}')
+        with h5py.File(filename, 'r') as f:
+            h = f['/Header']
+            if check:
+                assert int(h['Nspws'][()]) == 1  # not a hera file
+            if len(times) == 0:
+                info['freqs'] = h['freq_array'][()]
+                pol_array = h['polarization_array'][()]
+                npols = pol_array.size
+                pol_indices = {POL_NUM2STR_DICT[n]: cnt for cnt, n in enumerate(pol_array)}
+                info['pols'] = list(pol_indices.keys())
+            elif check:
+                assert int(h['Nfreqs'][()]) == info['freqs'].size
+            ntimes = int(h['Ntimes'][()])
+            times.append((h['time_array'][:ntimes], filename))
+            ant1_array = h['ant_1_array'][::ntimes]
+            ant2_array = h['ant_2_array'][::ntimes]
+            _hash = hash((ant1_array.tobytes(), ant2_array.tobytes()))
+            # map baselines to array indices for each unique antenna order
+            if _hash not in inds:
+                inds[_hash] = {(i, j): slice(n * ntimes, (n + 1) * ntimes)
+                               for n, (i, j) in enumerate(zip(ant1_array,
+                                                              ant2_array))}
+                if 'bls' not in info:
+                    info['bls'] = set(inds[_hash].keys())
+                else:
+                    info['bls'].intersection_update(set(inds[_hash].keys()))
```

Why are we doing an intersection update? This seems very strange. Shouldn't this code error if the ant_1_array and ant_2_array don't match from file to file?

This seems like it could silently drop baselines not common to all files.

---

I don't see why this would need to error. If baselines legitimately change from file to file, then the only thing "read all" could mean is to return the baselines which appear in all files. If you provide specific baselines, and they are not in all the files, it will error out in that case.

---

In that case, can you clarify that in the docstring?